### PR TITLE
Bugfix/#2907 warn popup and habit statistics tests

### PR DESCRIPTION
**Pull request by dyadyushko**
warn popup and habit stat service unit tests

### CODE REVIEW

**Review by Anastasiia-Hevchuk — APPROVED** [review]

**Review by Edw1x — APPROVED** [review]

File: src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
```
Comment (JimHolman1199) [inline src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.spec.ts:2]: remove empty space

File: src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      expect(component.popupTitle).toBe(popupDataStub.popupTitle);
+      expect(component.popupSubtitle).toBe(popupDataStub.popupSubtitle);
+      expect(component.popupCancel).toBe(popupDataStub.popupCancel);
+      expect(component.popupConfirm).toBe(popupDataStub.popupConfirm);
+    });
+
+    it('should execute close method inside ngOnDestroy', () => {
+      const spy = spyOn(component.matDialogRef, 'close');
+
```
Comment (JimHolman1199) [inline src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.spec.ts:79]: let's have only 1 empty space before expect for each 'it'

File: src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts [hunk 3]
```diff
@@ -6,17 +6,17 @@ import { takeUntil } from 'rxjs/operators';
...
-  styleUrls: ['./warning-pop-up.component.scss'],
+  styleUrls: ['./warning-pop-up.component.scss']
 })
 export class WarningPopUpComponent implements OnInit, OnDestroy {
   public popupTitle: string;
   public popupSubtitle: string;
   public popupConfirm: string;
   public popupCancel: string;
   public closeButton = './assets/img/profile/icons/cancel.svg';
-  private destroyed$: ReplaySubject<any> = new ReplaySubject<any>(1);
```
Comment (JimHolman1199) [inline src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts:17]: there is no need to remove private
use // @ ts-ignore instead

File: src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts [hunk 4]
```diff
@@ -6,17 +6,17 @@ import { takeUntil } from 'rxjs/operators';
...
+  destroyed$: ReplaySubject<any> = new ReplaySubject<any>(1);
 
-  constructor(private matDialogRef: MatDialogRef<WarningPopUpComponent>, @Inject(MAT_DIALOG_DATA) public data) {}
+  constructor(public matDialogRef: MatDialogRef<WarningPopUpComponent>, @Inject(MAT_DIALOG_DATA) public data) {}
```
Comment (JimHolman1199) [inline src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts:19]: same

File: src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts [hunk 5]
```diff
@@ -37,14 +37,14 @@ export class WarningPopUpComponent implements OnInit, OnDestroy {
       .subscribe(() => this.userReply(false));
   }
 
-  private setTitles(): void {
+  setTitles(): void {
```
Comment (JimHolman1199) [inline src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts:40]: the same

File: src/app/main/service/habit-statistic/habit-statistic.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,142 @@
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+
+import { HabitStatisticService } from './habit-statistic.service';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { of } from 'rxjs';
+import { HabitDto } from '@global-models/habit/HabitDto';
+import { AvailableHabitDto } from '@global-models/habit/AvailableHabitDto';
+import { NewHabitDto } from '@global-models/habit/NewHabitDto';
+
+fdescribe('HabitStatisticService', () => {
```
Comment (JimHolman1199) [inline src/app/main/service/habit-statistic/habit-statistic.service.spec.ts:10]: remove f before describe

File: src/app/main/service/habit-statistic/habit-statistic.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,142 @@
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+
```
Comment (JimHolman1199) [inline src/app/main/service/habit-statistic/habit-statistic.service.spec.ts:2]: remove an empty space

File: src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.spec.ts (hunk 1 above)
Comment (dyadyushko) [reply src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.spec.ts]: done

File: src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.spec.ts (hunk 2 above)
Comment (dyadyushko) [reply src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.spec.ts]: done

File: src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts (hunk 3 above)
Comment (dyadyushko) [reply src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts]: done

File: src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts (hunk 4 above)
Comment (dyadyushko) [reply src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts]: done

File: src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts (hunk 5 above)
Comment (dyadyushko) [reply src/app/main/component/shared/components/warning-pop-up/warning-pop-up.component.ts]: done

File: src/app/main/service/habit-statistic/habit-statistic.service.spec.ts (hunk 6 above)
Comment (dyadyushko) [reply src/app/main/service/habit-statistic/habit-statistic.service.spec.ts]: done

File: src/app/main/service/habit-statistic/habit-statistic.service.spec.ts (hunk 7 above)
Comment (dyadyushko) [reply src/app/main/service/habit-statistic/habit-statistic.service.spec.ts]: done

**Review by JimHolman1199 — APPROVED** [review]

**Pull request merged**